crates/fuel-core/src/graphql_api/ports.rs

```diff
@@ -336,6 +344,14 @@ pub mod worker {
         fn block_events(&self) -> BoxStream<SharedImportResult>;
     }
 
+    pub trait ImportResultProvider: Send + Sync {
```

Comment:
I'm not a fan of the name `ImportResultProvider` here or on the adapter. It is referring to the type `ImportResult` which only has meaning inside the block importer context.

Why not just call it `ImportBlock` or something to that effect?

Reply:
Is it even appropriate for it to be using the importer type here? Looking at the impl, we are constructing the `SharedImportResult`.

Reply:
The second question partially answers the first one=D The block importer returns `SharedImportResult`, which we work with within the off-chain worker.

I will merge Block Importer port with ImportResultProvider